File: IPWorks ZIP Samples/Open SevenZip/c++/opensevenzip.cpp
/*
 * IPWorks ZIP 2024 C++ Edition - Sample Project
 *
 * This sample project demonstrates the usage of IPWorks ZIP in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/ipworkszip
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */

#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include "../../include/ipworkszip.h"
#define LINE_LEN 100

int main()
{
	int ret_code = 0;
	SevenZip sevenzip;
	char buffer[LINE_LEN + 1];

	printf("**********************************************************\n");
	printf("* This is a demo of the SevenZip component               *\n");
	printf("**********************************************************\n\n");

	printf("Please enter the name of the archive to extract [samplezip.7z]: ");
	fgets(buffer, LINE_LEN, stdin);
	buffer[strlen(buffer) - 1] = '\0';
	if (strlen(buffer) == 0) strcat(buffer, "samplezip.7z");
	sevenzip.SetArchiveFile(buffer);

	printf("Please enter the path to which the files will be extracted [. (current dir)]: ");
	fgets(buffer, LINE_LEN, stdin);
	buffer[strlen(buffer) - 1] = '\0';
	if (strlen(buffer) == 0) strcat(buffer, ".");
	sevenzip.SetExtractToPath(buffer);

	printf("\nExtracting ...\n\n");
	ret_code = sevenzip.ExtractAll();

	if (!ret_code)
		printf("Completed.\n\n");
	else
		printf("Error extracting: [%i] %s\n", ret_code, sevenzip.GetLastError());

	fprintf(stderr, "\npress <return> to continue...\n");
	getchar();

	return ret_code;
}

File: IPWorks ZIP Samples/Open SevenZip/php/open_sevenzip.php
<?php
/*
 * IPWorks ZIP 2024 PHP Edition - Sample Project
 *
 * This sample project demonstrates the usage of IPWorks ZIP in a 
 * simple, straightforward way. It is not intended to be a complete 
 * application. Error handling and other checks are simplified for clarity.
 *
 * www.nsoftware.com/ipworkszip
 *
 * This code is subject to the terms and conditions specified in the 
 * corresponding product license agreement which outlines the authorized 
 * usage and restrictions.
 */
require_once('../include/ipworkszip_sevenzip.php');
require_once('../include/ipworkszip_const.php');
?>
<?php
if ($argc < 2) {
  echo "Usage: php open_sevenzip.php inputfile\n\n";
  echo "  inputfile       the path to the 7z archive\n";
  echo "\nExample: php open_sevenzip.php samplezip.7z\n\n";
} else {
  try {
    $opensevenzip = new IPWorksZip_SevenZip();

    $opensevenzip->setArchiveFile($argv[1]);

    echo "Type \"?\" or \"help\" for a list of commands.\n";
    echo "opensevenzip> ";
    $command = "";
    $arguments = [];

    while (true) {
      $command = readline();
      $arguments = explode(" ", $command);

      if ($arguments[0] == "?" || $arguments[0] == "help") {
        echo "Commands: \n";
        echo "  ?                                   display the list of valid commands\n";
        echo "  help                                display the list of valid commands\n";
        echo "  scan                                scan the compressed archive\n";
        echo "  extract <files> <path>              extract the specified files to the specified path (separate multiple files with |)\n";
        echo "  extractall <path>                   extract all files in the archive to the specified path\n";
        echo "  quit                                exit the application\n";
      } else if ($arguments[0] == "scan") {
        $opensevenzip->doScan();
        for ($i = 0; $i < $opensevenzip->getFileCount(); $i++) {
          echo $opensevenzip->getFileCompressedName($i), "\n";
        }
      } else if ($arguments[0] == "extract") {
        if (count($arguments) > 2) {
          $opensevenzip->setExtractToPath($arguments[2]);
          $opensevenzip->doExtract($arguments[1]);
        }
      } else if ($arguments[0] == "extractall") {
        if (count($arguments) > 1) {
          $opensevenzip->setExtractToPath($arguments[1]);
          $opensevenzip->doExtractAll();
        }
      } else if ($arguments[0] == "quit" || $arguments[0] == "exit") {
        break;
      } else if ($arguments[0] == "") {
        // Do nothing.
      } else {
        echo "Invalid command.\n";
      } // End of command checking.

      echo "opensevenzip> ";
    }
  } catch (Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
  }
}
?>